DrivingSchoolPRA/app/Controllers/InstructorController.cs

```diff
@@ -76,6 +85,81 @@ public ActionResult<InstructorDto> Get(int id)
             }
         }
 
+        [HttpGet("[action]")]
+        public ActionResult<IEnumerable<InstructorDto>> GetByName(string filter)
+        {
+            try
+            {
+                if (!_context.Instructors.Any(i => i.Person.FirstName.StartsWith(filter)))
+                {
+                    return StatusCode(404, "No avaliable instructors.");
+                }
+
+                var instructors = _context.Instructors
+                    .Include(i => i.Person)
+                    .Include(i => i.Vehicle)
+                    .Where(i => i.Person.FirstName.StartsWith(filter))
+                    .Select(i => new InstructorDto
+                    {
+                        Id = i.PersonId,
+                        FirstName = i.Person.FirstName,
+                        Lastname = i.Person.Lastname,
+                        Email = i.Person.Email,
+                        Vehicle = i.Vehicle,
+                        Rating = CalculateRating(i.Idinstructor)
+                    });
+
+                return Ok(instructors);
+            }
+            catch (Exception e)
+            {
+                return StatusCode(500, e.Message);
+            }
+        }
+
+        [HttpGet("[action]")]
+        public ActionResult<IEnumerable<InstructorDto>> GetByRating(float min, float max)
+        {
+            try
+            {
+                if (min < 1 || max > 5)
+                {
+                    return BadRequest("Rating must be in range 1 - 5.");
+                }
+
+                var instructors = _context.Instructors
+                    .Include(i => i.Person)
+                    .Include(i => i.Vehicle)
+                    .Select(i => new InstructorDto
+                    {
+                        Id = i.PersonId,
+                        FirstName = i.Person.FirstName,
+                        Lastname = i.Person.Lastname,
+                        Email = i.Person.Email,
+                        Vehicle = i.Vehicle,
+                        Rating = CalculateRating(i.Idinstructor)
+                    })
+                    .Where(i => i.Rating >= min);
```

Comment:
TU bi vjv trebalo dodati && <= max

Reply:
bi al neki cudni linq exception baca tu pa to jos nije gotovo, al da treba

Reply:
ok